packages/credential-providers/src/fromAwsCliV2CompatibleProviderChain.ts

```diff
@@ -0,0 +1,77 @@
+import { remoteProvider } from "@aws-sdk/credential-provider-node/src/remoteProvider";
+import { createCredentialChain } from "@aws-sdk/credential-providers";
+import type { RuntimeConfigAwsCredentialIdentityProvider } from "@aws-sdk/types";
+import type { AwsCredentialIdentity } from "@aws-sdk/types";
+import { CredentialsProviderError } from "@smithy/property-provider";
+
+interface AwsCliV2CompatibleProviderOptions extends Partial<AwsCredentialIdentity> {
+  profile?: string;
+  logger?: Console;
+}
+
+/**
+ * Custom AWS CLI V2 Compatible Credential Provider Chain.
+ * Uses dynamic imports and `createCredentialChain` to mimic AWS CLI V2 behavior.
+ */
+export const fromAwsCliV2CompatibleProviderChain =
+  (_init: AwsCliV2CompatibleProviderOptions = {}): RuntimeConfigAwsCredentialIdentityProvider =>
+  async ({ callerClientConfig } = {}): Promise<AwsCredentialIdentity> => {
+    // Merge init with caller's client config (profile/region).
+    const init: AwsCliV2CompatibleProviderOptions = {
+      ..._init,
+      ...callerClientConfig,
+      logger: (_init.logger ?? callerClientConfig?.logger ?? console) as Console,
+    };
+
+    init.logger?.debug("@aws-sdk/custom-credential-chain - Initializing credential chain");
+
+    const { profile, ...awsCredentials } = init;
+
+    // 1. If credentials are explicitly provided, return them.
+    if (awsCredentials.accessKeyId && awsCredentials.secretAccessKey) {
+      init.logger?.debug("@aws-sdk/custom-credential-chain - Using credentials from constructor");
```

Comment:
```suggestion
      init.logger?.debug("@aws-sdk/credential-providers - fromAwsCliV2CompatibleProviderChain - using static credentials from initialization");
```

the prefix is the package name.